### PR TITLE
XW-3761 | Fix GlobalTitle::newMainPage

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/XW-3761

GlobalTitle::newMainPage uses Mediawiki:mainpage message to create mainpage url

@Wikia/x-wing

### CODE REVIEW

File: includes/wikia/GlobalTitle.php [hunk 1]
```diff
@@ -100,8 +100,13 @@ public static function newMainPage( $city_id = null ) {
 			throw new \Exception( 'Invalid $city_id.' );
 		}
 
-		// sure hope this redirects for the most part
-		$title = self::newFromText( 'Main Page', NS_MAIN, $city_id );
+		$title = self::newFromText( wfMsgForContent('mainpage'), NS_MAIN, $city_id );
```
Comment (WikiaTech) [inline includes/wikia/GlobalTitle.php]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by rogatty — APPROVED** [review]

**Review by hakubo — APPROVED** [review]

**Review by hakubo — APPROVED** [review]

**Review by rogatty — APPROVED** [review]

**Review by rogatty — APPROVED** [review]

**Review by rogatty — APPROVED** [review]

File: includes/wikia/GlobalTitle.php [hunk 2]
```diff
@@ -314,6 +320,8 @@ public function getFullURL( $query = '', $variant = false ) {
 			} );
 
 			$titleText = wfUrlencode( $localName );
+		} elseif ( $this->mUrlform === 'Main_Page' ) {
+			$titleText = wfMessage( 'mainpage' )->inLanguage( $this->mLang )->useDatabase( false )->plain();
```
Comment (katanka) [inline includes/wikia/GlobalTitle.php:324]: Based on the suggestion from @TK-999

File: includes/wikia/GlobalTitle.php [hunk 3]
```diff
@@ -100,8 +100,14 @@ public static function newMainPage( $city_id = null ) {
 			throw new \Exception( 'Invalid $city_id.' );
 		}
 
-		// sure hope this redirects for the most part
-		$title = self::newFromText( 'Main Page', NS_MAIN, $city_id );
+		$mainPageName = self::newFromText( 'Mainpage', NS_MEDIAWIKI, $city_id )->getContent();
+		$title = self::newFromText( $mainPageName, NS_MAIN, $city_id );
```
Comment (KockaAdmiralac) [inline includes/wikia/GlobalTitle.php:104]: Just a small question, should main page being outside of main namespace ([example](http://kocka.wikia.com/wiki/MediaWiki:Mainpage)) be handled here?

**Pull request merged**